CosmicReco/inc/DriftFitUtils.hh

```diff
@@ -34,14 +34,16 @@ using namespace mu2e;
 
 namespace DriftFitUtils{
   	TrackEquation ConvertFitToDetectorFrame(TrackAxes axes, XYZVec Position, XYZVec Direction);
-	double GetTestDOCA(Straw const& straw, double a0, double a1, double b0, double b1);
-        int GetAmbig(Straw const& straw, double a0, double a1, double b0, double b1);
-  	double GetPropVelocity(StrawResponse rep, ComboHit chit); 
-	double GetPropTime(ComboHit chit, Straw straw, double vprop);
-  	double TimeResidualTrans(Straw const&  straw, double doca, StrawResponse srep, double t0, ComboHit chit);
-  	double TimeResidualLong(Straw const&  straw, double doca, StrawResponse srep, double t0, ComboHit chit);
-  	double TimeResidual(Straw const&  straw, double doca, StrawResponse srep, double t0, ComboHit hit);
-  	double T0(Straw const&  straw, double doca, StrawResponse srep, double t0, ComboHit hit, double average);
+	double GetTestDOCA(ComboHit const& chit,XYZVec pos, XYZVec dir,  const Tracker* tracker);
```

Comment:
I believe that the arguments pos and dir are input only arguments, right?  If so the arguments should be XYZVec const& to avoid an unnecessary copy.   You will only need to change 2 lines: the function declaration and the function implementation - you won't need to change any of the calling code.

The same comments for  the overload of GetAmbig that use XYZVEC.

The general guidance is:

1) Maximize the use of const - it never hurts and it's surprising to me how often the compiler can take advantage of it.
2) If an argument is const, and the type is bigger than a pointer, then pass the argument by reference or pointer, not by value.

If rule 1) causes problems because other peoples' code is not const correct, ask that it be fixed or volunteer to fix it.

Reply:
yes I think this option is not used in many places, its the other variant of the function I use mostly. I will change this one though for future.